flatsurf/geometry/surface.py

```diff
@@ -758,11 +812,21 @@ def polygon(self, lab):
         try:
             data = self._p[lab]
         except IndexError:
-            raise ValueError("No known polygon with provided label "+str(lab)+". "+\
-                "This can be caused by failing to explore your surface. "+\
-                "See the documentation in flatsurf.geometry.surface.Surface_list.")
+            if self._reference_surface is None:
+                raise ValueError(f"No polygon with label {lab}.")
+
+            for label in self.label_iterator():
+                if label >= lab:
+                    break
+
+            if lab >= len(self._p):
+                raise ValueError(f"No polygon with label {lab}.")
```

Comment:
could you change the error message to `"no polygon with label {lab}"` (no upper case and no point)

Reply:
```suggestion
                raise ValueError(f"no polygon with label {lab}")
```